src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,81 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.StringJoiner;
+
 public class WorkWithFile {
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = this.readFromFile(fromFileName);
+        String report = this.createReport(dataFromFile);
+        this.writeToFile(report, toFileName);
+    }
+
+    private String readFromFile(final String fromFileName) {
+        File file = new File("./" + fromFileName);
+        StringJoiner result = new StringJoiner("|");
+
+        try (
+                FileReader fileReader = new FileReader(file);
+                BufferedReader bufferedReader = new BufferedReader(fileReader)
+
+        ) {
+            String nextString = bufferedReader.readLine();
+            while (nextString != null) {
+                result.add(nextString);
+                nextString = bufferedReader.readLine();
+            }
+
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file: File not Found", e);
+        }
+        return result.toString();
+    }
+
+    private String createReport(final String dataFromFile) {
+        StringBuilder result = new StringBuilder();
+        int supply = 0;
+        int buy = 0;
+        String[] dataFromFileArray = dataFromFile.split("\\|");
```

Comment:
```suggestion
        String[] dataFromFileArray = dataFromFile.split("\n");
```

Reply:
readFromFile method - StringJoiner result = new StringJoiner("|");

Reply:
Done
I also I needed to add "result.append(System.lineSeparator());